conans/test/util/tools_test.py

```diff
@@ -710,6 +710,38 @@ def vcvars_echo_test(self):
             self.assertIn("Conan:vcvars already set", str(output))
             self.assertIn("VS140COMNTOOLS=", str(output))
 
+    @unittest.skipUnless(platform.system() == "Windows", "Requires Windows")
+    def vcvars_env_not_duplicated_path_test(self):
+        """vcvars is not looking at the current values of the env vars, with PATH it is a problem because you
+        can already have set some of the vars and accumulate unnecessary entries."""
+        settings = Settings.loads(default_settings_yml)
+        settings.os = "Windows"
+        settings.compiler = "Visual Studio"
+        settings.compiler.version = "15"
+        settings.arch = "x86"
+        settings.arch_build = "x86_64"
+
+        # Set the env with a PATH containing the vcvars paths
+        tmp = tools.vcvars_dict(settings, only_diff=False)
+        tmp = {key.lower(): value for key, value in tmp.items()}
+        with tools.environment_append({"path": ";".join(tmp["path"])}):
+            previous_path = os.environ["PATH"].split(";")
+            # Duplicate the path, inside the tools.vcvars shouldn't have repeated entries in PATH
+            with tools.vcvars(settings):
+                path = os.environ["PATH"].split(";")
+                new_values = []
+                repeated_keys = []
+                for i in path:
+                    if i not in new_values:
+                        new_values.append(i)
+                    else:
+                        repeated_keys.append(i)
+
```

Comment:
I'm not sure this is working as expected. We are appending a `path` (lowercase) variable to `os.environ` and then we are working with the `PATH` (uppercase) one. I think that I need here, at least, a `self.assertTrue(len(repeated_keys) != 0)` check.

Reply:
`os.environ` is not case sensitive.